syntax = "proto3";


message all_reduce_msg {
  int32 client_rank = 1;
  bytes msg = 2;
}

message batch_msg {
  int32 client_rank = 1;
  bytes msg = 2;
  repeated int32 group_keys = 3;
  repeated bytes res = 4;
}

service MIService {
  rpc sum_enc_all_reduce (all_reduce_msg) returns (all_reduce_msg);
  rpc sum_enc_batch(batch_msg) returns (batch_msg);
}